src/hivpy/demographics.py

```diff
@@ -193,3 +229,27 @@ def initialise_age(self, count):
             age_distribution = ContinuousAgeDistribution.select_model(self.params['inc_cat'])
 
         return age_distribution.gen_ages(count)
+
+    def determine_deaths(self, population_data: pd.DataFrame) -> pd.Series:
+        """Get which individuals die in a time step, as a boolean Series."""
+        all_died = pd.Series([False] * len(population_data))
+        for sex in SexType:
+            for (low_age, high_age), rate in self.params["death_rates"][sex].items():
+                # Find everyone in the relevant age range
+                index = selector(population_data, sex=(operator.eq, sex),
+                                 age=(between, (low_age, high_age)))
+                group = population_data[index]
+                # Probability of dying, assuming time step of 3 months
+                prob_of_death = 1 - exp(-rate / 4)
```

Comment:
Can you explain this probability? This doesn't look correct to me

Reply:
I was copying the relevant formula from the old model (involving `ac_death_rate`). The division by 4 is assuming a 3 month interval, because the overall death rate refers to a whole year. I think I tried to derive it but wasn't sure exactly what "rate" this was, so I left it as-is. Worth clarifying for sure, I'll check again in the model description.

Reply:
Okay I have tried the calculation out and I think that this is correct if the original probability for a year is `1-exp(-rate)`, based ensuring that the probability of dying in a year is the same as the sum of the probabilities of:

- dying in the first period
- surviving first period and dying in second
- surviving first two and dying in third
- etc.

then we have basically a constant (probability of dying in a period) multiplying a geometric sum (probability of surviving a period to the power of the number of periods they survive), which ends up giving me a formula for probability of dying in a year divided by N:
P_N = 1 - (1 - P_year)^(1/N)
which all works out fine.